0006/README.md

```diff
@@ -0,0 +1,49 @@
+---
+id: 0006
+title: Resiliency in Tinkerbell Provisioner.
+status: Ideation
+authors: Chitrabasu Khare <chitrabasukhare89@gmail.com>
+---
+
+## Summary:
+
+This RFD proposes to have a resilient framework for Tinkerbell Provisioner.
+
+
+## Goals:
+
+- Bring resiliency in Tinkerbell. 
+- Tinkerbell as a setup should be highly available.
+
+## Content: 
+
+- We would like to have a framework where Tinkerbell as a solution is fault-tolerant.
+- All most all of the core services are managed by provisioner. So, it will be good to have a framework where provisioner is highly available.
+- It can be a multinode provisioner solution with active-active or active-passive implementation.
+- The Provisioner services should be partition tolerant during network failures or communication failure between provisioner nodes.
+- Additional logic would be required to sync services running inside provisioner nodes.
+- A separate machine to place an L3/L7 Load balancer or proxy would be required.
+
+
+## Problems and Suggestions:
+
+1) How does worker communicate with provisioner?
+-  The worker will communicate to the load balancer (or proxy) and the request will be routed to a replica of provisioner.
+-  Sessions need to be maintained inside Load balancer so that different requests of worker goes to the same replica of provisioner. 
+-  How sessions can be maintained with the latest changes of GRPC streams, requires some investigation. 
```

Comment:
As of now, most of micro-services are exposed to the worker which it uses for workflow execution, like boots, tik-server etc. In that case, session needs to be maintained by at least those services. 
I am yet to conclude this point. 
I have earlier suggested exposing all the tink-services via a reverse proxy. If we go via that way, we may need to only have session retained on the microservice who is exposing the tinkerbell IP, it could be nginx.